### PR TITLE
add name validation to avoid blank name submissions and special chara…

### PR DESCRIPTION
**Pull request by bijoysijo**
…cters in name

Fixes:
Fixed user being able to submit the name field as blank while signing up.
Made exceptions for a set of characters a user can submit through the name field.

#### Describe the changes you have made in this PR -

validates :name, presence: true, format
checks if the name is present and allows only certain characters to be present in the name. i.e avoids name field being blank and names containing, %^$*# etc..

Found out that the tests in signup_spec.rb restricted these validations. updated tests to pass the validations tests made In this PR.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -31,7 +31,12 @@ class User < ApplicationRecord
   after_commit :create_members_from_invitations, on: :create
 
   has_attached_file :profile_picture, styles: { medium: "205X240#", thumb: "100x100>" }, default_url: ":style/Default.jpg"
+
+  #validations for user
```
[the remark this answers is not in the document]
Comment (bijoysijo) [reply app/models/user.rb]: updated

File: app/models/user.rb [hunk 2]
```diff
@@ -31,7 +31,12 @@ class User < ApplicationRecord
...
+
   validates_attachment_content_type :profile_picture, content_type: %r{\Aimage/.*\z}
+  validates :name, presence: true
+  validates_format_of :name, :with => /\A[^0-9`!@#\$%\^&*+_=]+\z/
```
[the remark this answers is not in the document]
Comment (bijoysijo) [reply app/models/user.rb]: updated

[the remark this answers is not in the document]
Comment (bijoysijo) [reply app/models/user.rb]: updated

**Comment by tachyons** [comment]
This validations will block non English names, also makes some of the existing users invalid

**Comment by bijoysijo** [comment]
> This validations will block non English names, also makes some of the existing users invalid

Can you provide an example name? so I can work on this better.

**Comment by bijoysijo** [comment]
Made a new PR #1646, addressing email validation and name field validation.

**Pull request closed without merging**